odoo/addons/base/models/ir_model.py

```diff
@@ -1925,26 +1925,26 @@ def xmlid_lookup(self, xmlid):
         return result
 
     @api.model
-    def xmlid_to_res_model_res_id(self, xmlid, raise_if_not_found=False):
+    def _xmlid_to_res_model_res_id(self, xmlid, raise_if_not_found=False):
         """ Return (res_model, res_id)"""
         try:
-            return self.xmlid_lookup(xmlid)[1:3]
+            return self._xmlid_lookup(xmlid)[1:3]
         except ValueError:
             if raise_if_not_found:
                 raise
             return (False, False)
 
     @api.model
-    def xmlid_to_res_id(self, xmlid, raise_if_not_found=False):
+    def _xmlid_to_res_id(self, xmlid, raise_if_not_found=False):
         """ Returns res_id """
-        return self.xmlid_to_res_model_res_id(xmlid, raise_if_not_found)[1]
+        return self._xmlid_to_res_model_res_id(xmlid, raise_if_not_found)[1]
 
     @api.model
-    def xmlid_to_object(self, xmlid, raise_if_not_found=False):
+    def _xmlid_to_object(self, xmlid, raise_if_not_found=False):
```

Comment:
Honestly, I believe it would be a good opportunity to rename those methods, since we loose retro-compatibility anyway

Reply:
The advantage is that it's easy to migrate. Error "Method `xmlid_to_res_id` does not exists" -> grep -> find `_xmlid_to_res_id` problem solved.